internal/provider/resources/common_cluster.go

```diff
@@ -0,0 +1,47 @@
+package resources
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+
+	"github.com/astronomer/terraform-provider-astro/internal/clients"
+	"github.com/astronomer/terraform-provider-astro/internal/clients/platform"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/retry"
+)
+
+// resourceRefreshFunc returns a retry.StateRefreshFunc that polls the platform API for the cluster status
+// If the cluster is not found, it returns "DELETED" status
+// If the cluster is found, it returns the cluster status
+// If there is an error, it returns the error
+// WaitForStateContext will keep polling until the target status is reached, the timeout is reached or an err is returned
+func ResourceRefreshFunc(ctx context.Context, platformClient *platform.ClientWithResponses, organizationId string, clusterId string) retry.StateRefreshFunc {
```

Comment:
we should use this in `cluster_resource` and remove the duplicate function in that one to use this one